echopop/utils/batch_load.py

```diff
@@ -0,0 +1,608 @@
+
+    # Get region info
+    # ---- Region filenames
+    region_files = configuration_dict["export_regions"]
+    # ---- Region names
+    region_names = export_settings["regions"]
+    # ---- Root directory
+    root_dir = configuration_dict["data_root_dir"]
+    # ---- Read in region definitions
+    regions_df = load_export_regions(region_files, region_names, root_dir)
+
+    # Read in and consolidate all export files (intervals, layers, cells)
+    transect_data, interval_template = consolidate_exports(
+        transect_reference, export_settings["max_transect_spacing"]
+    )
+    # ---- Rename a column
+    interval_template = interval_template.rename(columns={"max_depth": "bottom_depth"})
+
+    # Read in the haul-to-transect KS strata key
+    haul_transect = get_haul_transect_key(configuration_dict, root_dir)
+
+    # Vertical/areal integration (over groups)
+    for key, values in region_names.items():
+        # ---- Get valid region names
+        region_filter = values
+        # ---- Apply the region filter
+        transect_regions = filter_export_regions(
+            transect_data,
+            index_variable=index_variable,
+            region_filter=region_filter,
+            unique_region_id=unique_region_id,
+            region_class_column=region_class_column,
+            impute_regions=True,
+        )
+        # ---- Calculate transect spatial metrics
+        transect_layer_summary = export_transect_layers(transect_regions)
+        # ---- Vertically sum backscatter
+        nasc_intervals = (
+            transect_regions.groupby(list(index_variable + [unique_region_id]))["nasc"]
+            .sum()
+            .to_frame("NASC")
+            .reset_index()
+        )
+        # ---- Partition specific grouped regions
+        grouped_region = regions_df[regions_df["group"] == key]
+        # ---- Merge stratum and haul information into the integrated NASC dataframe
+        nasc_haul_strata = group_merge(nasc_intervals, [grouped_region, haul_transect])
+        #
+        full_interval_df = interval_template.merge(nasc_haul_strata, how="left")
+        #
+        full_interval_strata_df = full_interval_df.merge(transect_layer_summary, how="left")
+        # ---- Sort
+        full_interval_strata_df = full_interval_strata_df.sort_values(
+            ["transect_num", "vessel_log_start", "vessel_log_end"]
+        )
+        # ---- Fill NaN region id column with 999
+        full_interval_strata_df[unique_region_id] = (
+            full_interval_strata_df[unique_region_id].fillna(999).astype(int)
+        )
+        # ---- Fill haul with 0's
+        full_interval_strata_df["haul_num"] = (
+            full_interval_strata_df["haul_num"].fillna(0).astype(int)
+        )
+        # ---- Fill stratum with 1's
+        full_interval_strata_df["stratum_num"] = (
+            full_interval_strata_df["stratum_num"].fillna(1).astype(int)
+        )
```

Comment:
Just wondering why you fill missing stratum with 1s. Is it because stratum=1 does not get processed?

Reply:
`stratum=1` gets processed. This just comports with what Chu had written for `EchoPro`, so I wanted to maintain that consistency across files.